feat: Interactive tutorials code samples for search service

t-karasova opened the pull request:
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [ ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/nodejs-retail/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [ ] Ensure the tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

Fixes #<issue_number_goes_here> 🦕

t-karasova committed:
feat: Code samples and tests for Retail Search Service are added

feat: add mocha config and change package.json

removed eccessive folder

fix: code review fixes

fix: code review fixes

t-karasova committed:
feat: Code samples and tests for Retail Search Service are added

feat: add mocha config and change package.json

t-karasova committed:
feat: Interactive tutorials code samples for search service

t-karasova commented:
@bcoe 
Hi Benjamin, 
I have created this pull request to add the code samples for Interactive Tutorials for Retail API. 
You have already seen them in https://github.com/googleapis/nodejs-retail/pull/119, which now is closed, it was too big.

The code samples are moved to the folder "samples/interactive-tutorials" to separate them from the code samples, which are used for displaying in the static documentation pages. 
It was decided, that Our tutorials + code samples will be shown only by clicking the button "Guide me" in the Retail documentation page and will be opened in the Cloud Shell IDE. 

In this PR I have created the code samples for Retail Search service and tests.

I have modified .kokoro configs to read the secrets from "retail-samples-secrets" secret manager.

For the tests we use our own Cloud project with preset catalog data. And I would like to ask you how should I add the service account token to our project to authorize tests?
 
 Thank you

t-karasova committed:
Merge branch 'main' into retail-tutorials-search

chenlei1216 commented on samples/interactive-tutorials/search/search_simple_query.js:
```diff
@@ -0,0 +1,77 @@
+  const {SearchServiceClient} = require('@google-cloud/retail');
+
+  const projectNumber = process.env['PROJECT_NUMBER'];
+  const apiEndpoint = 'retail.googleapis.com';
```
nit: this can be omitted

t-karasova replied:
done

chenlei1216 commented on samples/interactive-tutorials/search/search_with_pagination.js:
```diff
@@ -0,0 +1,94 @@
+  };
+
+  // Get next page token from the response
+  const getNextPageToken = response => {
```
Should this be in the above block, or after callSearch()?

t-karasova replied:
done

gcf-owl-bot[bot] committed:
chore: update v2.12.0 gapic-generator-typescript (#132)

- [ ] Regenerate this pull request now.

Committer: @summer-ji-eng
PiperOrigin-RevId: 424244721

Source-Link: https://github.com/googleapis/googleapis/commit/4b6b01f507ebc3df95fdf8e1d76b0ae0ae33e52c

Source-Link: https://github.com/googleapis/googleapis-gen/commit/8ac83fba606d008c7e8a42e7d55b6596ec4be35f
Copy-Tag: eyJwIjoiLmdpdGh1Yi8uT3dsQm90LnlhbWwiLCJoIjoiOGFjODNmYmE2MDZkMDA4YzdlOGE0MmU3ZDU1YjY1OTZlYzRiZTM1ZiJ9

kweinmeister committed:
Merge branch 'main' into retail-tutorials-search

gcf-owl-bot[bot] committed:
🦉 Updates from OwlBot

See https://github.com/googleapis/repo-automation-bots/blob/main/packages/owl-bot/README.md

kweinmeister committed:
Merge branch 'main' into retail-tutorials-search

Commit:
Merge branch 'main' into retail-tutorials-search

t-karasova committed:
fix: change boost condition for search

kweinmeister committed:
Merge branch 'main' into retail-tutorials-search

gcf-owl-bot[bot] committed:
🦉 Updates from OwlBot

See https://github.com/googleapis/repo-automation-bots/blob/main/packages/owl-bot/README.md

tkarasyova committed:
kokoro config is changed

tkarasyova committed:
PROJECT_NUMBER changed to GOOGLE_CLOUD_PROJECT_NUMBER

chenlei1216 commented on samples/interactive-tutorials/search/search_simple_query.js:
```diff
@@ -0,0 +1,76 @@
+// Copyright 2021 Google Inc. All Rights Reserved.
```
2022 Google LLC

chenlei1216 commented on samples/interactive-tutorials/search/search_with_pagination.js:
```diff
@@ -0,0 +1,91 @@
+  // Call search
+  await callSearch();
+
+  //PASTE CALL WITH NEXT PAGE TOKEN HERE:
```
nit: move this outside the tag range?

t-karasova commented:
The PR is closed due to the CLA issue, the changes are duplicated in the #146 PR
